enhancements/sig-architecture/14-addon-cluster-proxy/README.md

```diff
@@ -0,0 +1,260 @@
+  the addon-framework.
+- Automate the certificate rotation of both proxy-servers and proxy-agents.
+
+### Non-Goals
+
+- Customizing the source code of apiserver-network-proxy.
+
+## Proposal
+
+#### Architecture
+
+
+The following figure shows the interaction between the components:
+
+```
+   
+                     -----------------------------------------
+                     | Hub-Cluster                           |
+                     |                                       |
+  ----------------   | ----------------     ---------------  |
+  | Proxy Client |> > >| Proxy Server | <===| Proxy Addon |  |
+  ----------------   | ----------------     |  Controller |  |
+                     |      V               ---------------  |
+                     |      V                    ‖           |
+                     |    / V \                  ‖           |
+                     ----/| V |\-----------------‖------------
+                          | V |                  ‖
+                   Reverse| V |Tunnel            ‖ Install/Operate
+                          | V |                  ‖ 
+ -------------------------| V |-------------     ‖
+ | Managed-Cluster        | V |            |     ‖
+ |                        | v |            |     ‖
+ | -----------------   ---------------     |     ‖
+ | | KubeApiserver |< <| Proxy Agent | <==========
+ | -----------------   ---------------     |
+ -------------------------------------------
+
+```
+
+- A proxy-agent that establishes a long connection towards the proxy-server
+  and register itself w/ the cluster's identity.
+- A proxy-server that receives handshakes from proxy-agent and maintains 
+  tunnel over the long connection.
+- A proxy-client requesting the target kube-apiserver through the tunnel
+  over either websocket HTTP-CONNECT or grpc protocol.
+- An addon controller automates the installation of proxy servers and proxy 
+  agents by consuming the corresponding configuration custom resource. And
+  optionally the proxy-agents can aggregate and report the status upwards
+  so that admins can checkout their healthiness easily w/o even touching
+  the managed-clusters.
+
+
+#### Components
+
+According to the latest design and implementation of [addon-framework](https://github.com/open-cluster-management-io/enhancements/tree/main/enhancements/sig-architecture/8-addon-framework),
+we will bring a new agent controller named `cluster-proxy-agent-controller` that
+qualifies the interfaces defined in addon-framework and automatically does the 
+following things upon discovering a new `ManagedClusterAddon` resource:
+
+1. Signing all the required X509 certificates and keys beforehand:
+   
+    i.   a CSR requesting server certificate which will be shared across the 
+         replicas of the proxy servers.
+   
+    ii.  a CSR requesting agent certificate which will be shared by the proxy 
+         agents.
+   
+    iii. a CSR requesting client certificate for logging onto in the tunnel.
```

Comment:
Will these CRSs be created and approved by user in advance? or the `cluster-proxy-agent-controller` will handle this?

Reply:
i think so, the agent CSRs should be auto-approved by the operator if (1) the managedcluster is present (2) the managedclusteraddon is present